### PR TITLE
Add gp_tablespace_path function to return tablespace path.

### PR DESCRIPTION
**Pull request by fanfuxiaoran**
gp_tablespace_path returns full tablespace path like
/tmp/foo/GPDB_8.4_301801031_db1 with param tablespace
location '/tmp/foo' and param dbid 1.

Author: Xiaoran Wang <xiwang@pivotal.io>
Author: Max Yang <myang@pivotal.io>

### CODE REVIEW

File: src/include/catalog/pg_proc.h [hunk 1]
```diff
@@ -2368,6 +2368,9 @@ DESCR("convert generic options array to name/value table");
 DATA(insert OID = 1619 (  pg_typeof				PGNSP PGUID 12 1 0 0 f f f f f s 1 0 2206 "2276" _null_ _null_ _null_ _null_  pg_typeof _null_ _null_ _null_ ));
 DESCR("returns the type of the argument");
 
+DATA(insert OID = 7305 (  gp_tablespace_path	PGNSP PGUID 12 1 0 0 f f f f f s 2 0 25 "25 23" _null_ _null_ _null_ _null_  gp_tablespace_path _null_ _null_ _null_ ));
```
Comment (hlinnaka) [inline src/include/catalog/pg_proc.h]: Please add this to pg_proc.sql instead. That's where most GPDB-specific functions are defined.

Comment (fanfuxiaoran) [reply src/include/catalog/pg_proc.h]: Thanks.  Updated it .

File: src/backend/utils/adt/misc.c [hunk 2]
```diff
@@ -487,3 +489,24 @@ pg_typeof(PG_FUNCTION_ARGS)
...
+ * pg_tablespace, and ending 1 is the dbid.
+ */
+Datum
+gp_tablespace_path(PG_FUNCTION_ARGS)
+{
+	char		tblspace_path[MAXPGPATH];
+	const char	*spclocation = text_to_cstring(PG_GETARG_TEXT_PP(0));
+	int32_t		dbid = PG_GETARG_INT32(1);
+	int			ret;
+	ret = snprintf(tblspace_path, sizeof(tblspace_path),
```
Comment (ashwinstar) [inline src/backend/utils/adt/misc.c]: How do we make format here remains in-sync with`tablespace_version_directory()`, like someone accidentally changes `tablespace_version_directory()` but misses to change here.

**Comment by fanfuxiaoran** [comment]
Got it . Thanks!

**Pull request merged**